structure/parser.go

```diff
@@ -81,3 +82,20 @@ type ArrayParser interface {
 	WithReferenceArrayParser(reference int) ArrayParser
 	WithReferenceErrorReporter(reference int) ErrorReporter
 }
+
+// ForgivingTimeString is a helper function added specifically to handle https://tidepool.atlassian.net/browse/BACK-1161
+// It should be deprecated once Dexcom fixes their API.
+func ForgivingTimeString(stringValue string) (forgivingTime string) {
+	if len(stringValue) < 19 {
+		forgivingBytes := []byte("0000-01-01T00:00:00")
+		for i := range forgivingBytes {
+			if i < len(stringValue) {
```

Comment:
This condition will always be true

Reply:
Not always - For example, if `stringValue` is `2019-02-01T13:00`, then `len(stringValue) < 19` but `len(forgivingBytes) > len(stringValue)`.

Reply:
Right. I thought you were iterating over `stringValue` which would have made that condition unnecessary.